src/opm/input/eclipse/Schedule/Schedule.cpp

```diff
@@ -1243,6 +1244,20 @@ File {} line {}.)", pattern, location.keyword, location.filename, location.linen
         return WellMatcher(sched_state->well_order.get(), sched_state->wlist_manager.get());
     }
 
+    std::function<std::unique_ptr<SegmentMatcher>()>
+    Schedule::segmentMatcherFactory(const std::size_t report_step) const
+    {
+        return {
+            [report_step, this]() -> std::unique_ptr<SegmentMatcher>
+            {
+                const auto ix = (report_step < this->snapshots.size())
+                    ? report_step
+                    : this->snapshots.size() - 1;
+
+                return std::make_unique<SegmentMatcher>(this->snapshots[ix]);
+            }
+        };
```

Comment:
> Why is the last snapshot correct if `report_step > this->snapshots.size()`?

Brainfart on my part.  There was a time, when we used the `TimeMap` class, that `report step == size()` was somewhat meaningful, but that has not been the case since we moved to the `ScheduleState` strategy in commits 37e29266e (PR #2310) and 37f07a1b1 (PR #2500).  I've pushed an update to use `Schedule::operator[]()` instead which will throw an exception if someone passes an out-of-bounds report step.

Reply:
Thanks for the explanations (so this at various other places) and for the update.